krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -157,23 +206,33 @@ func (r *itfce) SetNetworkInstanceName(s string) error {
 	return nil
 }
 
-// SetInterfaceSpec sets the spec attributes in the kubeobject
-func (r *itfce) SetInterfaceSpec(spec *nephioreqv1alpha1.InterfaceSpec) error {
+// SetSpec sets the spec attributes in the kubeObject
+func (r *itfce) SetSpec(spec *nephioreqv1alpha1.InterfaceSpec) error {
```

Comment:
Rather than nest everything in `if spec != nil` it's easier to read if you do:

```
 if spec == nil {
  return nil
}
```

Or maybe you want to return an error in that case? Either way - return early instead of nesting the whole body in an "if".

Reply:
yes agreed will fix. Was just hacking something together to agree on the approach going fwd as this is applicable to all libraries.

Reply:
fixed